gate/gate.go

```diff
@@ -159,3 +158,35 @@ func (g *blockingGate) Done() {
 		panic("gate.Done: more operations done than started")
 	}
 }
```

Comment:
Why was this introduced? This does exactly the same thing as the existing blocking gate?